### PR TITLE
Allowing users to choose a reddit thread instead of being randomized

### PR DESCRIPTION
**Pull request by DomizianoScarcelli**
Added a new .env variable that allows the user to insert a thread link

If the user inserts a thread link, it will be used to make the video, otherwise, the thread will be randomized as the previous version.

### CODE REVIEW

**Comment by DomizianoScarcelli** [comment]
Closes issues #237

**Comment by 5vl** [comment]
Hi, I'm not someone who is a reviewer or anything, but you might want to add .idea/ and .DS_STORE to the .gitignore

**Comment by DomizianoScarcelli** [comment]
> Hi, I'm not someone who is a reviewer or anything, but you might want to add .idea/ and .DS_STORE to the .gitignore

Yes, I definitely forgot about it, thanks 😄

**Review by TechStudent11 — APPROVED** [review]
Very useful!

File: reddit/subreddit.py [hunk 1]
```diff
@@ -35,24 +34,36 @@ def get_subreddit_threads():
         username=os.getenv("REDDIT_USERNAME"),
         password=passkey,
     )
+    # If the user inserts a thread link, pick that one
+    if os.getenv("THREAD_LINK"):
```
Comment (jacesleeman) [inline reddit/subreddit.py:38]: if os.getenv("THREAD_LINK"):

makes it look for THREAD_LINK in .env not THREAD

File: .env.template [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
 
+
+SUBREDDIT="askReddit"
+
 THEME=""
 
-SUBREDDIT=""
+# Link of the thread
+# (e.g. https://www.reddit.com/r/memes/comments/sbr31o/discordggrmemes_the_official_rmemes_discord_server/)
+THREAD=""
```
Comment (jacesleeman) [inline .env.template:17]: so this would need to be THREAD_LINK=""

File: reddit/subreddit.py [hunk 3]
```diff
@@ -35,24 +34,36 @@ def get_subreddit_threads():
...
 
-    if os.getenv("SUBREDDIT"):
-        subreddit = reddit.subreddit(os.getenv("SUBREDDIT"))
+        print_step(f"Getting the inserted thread...")
+
+        thread_id = os.getenv("THREAD_LINK").split("/")[6]
```
Comment (jacesleeman) [inline reddit/subreddit.py:42]: or just make this and     if os.getenv("THREAD_LINK"): just say THREAD

**Review by jacesleeman — CHANGES_REQUESTED** [review]

**Review by jacesleeman — APPROVED** [review]

**Review by EmreTech — COMMENTED** [review]
I think it would be better if instead the user is prompted to input a link for a thread instead of keeping it in the env file.

**Comment by DomizianoScarcelli** [comment]
> I think it would be better if instead the user is prompted to input a link for a thread instead of keeping it in the env file.

Yes maybe it has some more sense. I added a .env variable (RANDOM_THREAD) that accepts yes or no as values. If no (or empty) the script will ask the user for the link inside the terminal, otherwise the thread will be randomized.

**Review by jacesleeman — APPROVED** [review]

**Comment by DomizianoScarcelli** [comment]
@CallumIO Hi, can I ask for a review to merge this PR?

**Comment by jacesleeman** [comment]
make sure to edit the README checking off the thing

**Review by EmreTech — APPROVED** [review]

**Comment by elebumm** [comment]
Looks great. Just removed the unused imports.

**Review by elebumm — APPROVED** [review]

**Pull request merged**